### DISCUSSION TITLE
It downloads videos randomly. How do I get it to download in a specific order? Where on my hard drive are the videos going? I have a mac

### DISCUSSION THREAD

**Discussion by DeeDeeWright** [aclap-dev/video-downloadhelper#440] [Help and Q&A]
### Your question

t downloads videos randomly. How do I get it to download in a specific order? Where on my hard drive are the videos going? I have a mac 

### Addon & CoApp version and details

version: 8.2.2.9
channel: stable
build date: 2024-03-19
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-US
license: unset
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/Users/deborahwright",
  "converterBinary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "n6.1-11-gc30f360955"
}

**Comment by Season4episode10** [comment]
Even if added in order when concurrently adding several files they will finish according to their individual size and specific network conditions (as would occur with any binary download) if your concern is naming, learn how to use the "Side Download" option as this gives you the option to name them, they will be consecutively named according to the order you added each individual download regardless of what order they finish if you don't use the "Side Download" option.

To find the download folder, (answered this for another Mac user once before ... I'm not a Mac user but the program works the same on all OS's and all instructions are available for you to read)  click on the 3 ball icon, click the settings icon and look for "Default download directory" in the "Behavior" tab

**Reply by DeeDeeWright** [reply]
Oh my gosh! Thank you so much! That helped me greatly! I'm so happy now. I have been trying to figure this out for days. It worked exactly like you said. I haven't tried to download anything else yet though. Another quick question. You're saying I can't rename each video I can just dump them in a folder correct? 

What I was thinking though was that I can change folders each time I save a set of videos so I will know which one is which if I can't save each video.

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user.

VDH can download each video into a different directory, & you can give each video a name of your own choosing.  But you have to use the CoApp as your download processor.  You'll want to go into the VDH settings & open the Behavior tab.  The relevant settings are right at the top of that page.  While you're there, I would strongly recommend you enable the native file picker, another setting that you will see near the top of that page of settings.  Then when you download something, do a Download, not a Quick Download.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/6b881d34-db09-4d5b-9af1-f8d058a3dbb1)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/556746ef-13b9-4ea0-87ed-57b9a52628ae)
![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/78f98416-d234-4c2e-9217-c2263dc1c3e3)

To set the default action, do things in this order:

1.  Open the secondary menu like I show in those images.
2.  Click the indicated box at the bottom of the secondary menu.
3.  Click the Download option.

Forever after (until you change it again as I've just described) you won't need to hover then click to open the secondary menu.  You can just click the variant on the main VDH menu.  That will launch a Download function, whose first step is a file selection dialog.  That's where you can pick a directory & a file name for each download you perform.
